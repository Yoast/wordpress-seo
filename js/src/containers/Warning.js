import { withSelect } from "@wordpress/data";
import { Warning } from "@yoast/components";

export default withSelect( select => {
	const { getWarningMessage } = select( "yoast-seo/editor" );

	return { message: getWarningMessage() };
} )( Warning );
