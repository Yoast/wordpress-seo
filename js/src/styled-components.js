import * as styledComponents from "../../node_modules/styled-components";

window.yoast = window.yoast || {};
window.yoast.styledComponents = styledComponents;
