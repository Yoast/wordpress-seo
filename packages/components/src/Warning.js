import isEmpty from "lodash/isEmpty";
import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

/* Yoast dependencies */
import { colors } from "@yoast/style-guide";
import { getDirectionalStyle } from "@yoast/helpers";

// Internal dependencies.
import SvgIcon from "./SvgIcon";

const WarningBox = styled.div`
	display: flex;
	padding: 16px;
	background: ${ colors.$color_alert_warning_background };
	color: ${ colors.$color_alert_warning_text };
`;

const StyledSvgIcon = styled( SvgIcon )`
	margin-top: 2px;
`;

const MessageContainer = styled.div`
	margin: ${ getDirectionalStyle( "0 0 0 8px", "0 8px 0 0" ) };
`;

/**
 * A warning to show in the meta box.
 * (Consists of a warning icon and a text on a yellow background)
 *
 * @param {Array} message The warning message to display.
 */
class Warning extends React.Component {
	/**
	 * Renders the YoastWarning component.
	 *
	 * @returns {React.Element|null} The rendered YoastWarning.
	 */
	render() {
		const { message } = this.props;
		if ( isEmpty( message ) ) {
			return null;
		}
		return <WarningBox>
			<StyledSvgIcon icon="exclamation-triangle" size="16px" />
			<MessageContainer>
				{ message }
			</MessageContainer>
		</WarningBox>;
	}
}

Warning.propTypes = {
	message: PropTypes.array,
};

Warning.defaultProps = {
	message: [],
};

export default Warning;
