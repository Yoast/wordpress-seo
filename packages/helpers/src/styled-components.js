/**
 * Returns a function that returns the correct style based on if the current
 * theme is right to left.
 *
 * This is determined by the `isRtl` property in the styled components theme.
 *
 * @param {string} leftToRightStyle  Style to return if the theme is left to right.
 * @param {string} rightToLeftStyle Style to return if the theme is right to left.
 *
 * @returns {Function} A function that returns the right styled based on the
 *                       theme in the props.
 */
export function getDirectionalStyle( leftToRightStyle, rightToLeftStyle ) {
	return ( props ) => {
		return ( props.theme.isRtl ? rightToLeftStyle : leftToRightStyle );
	};
}
