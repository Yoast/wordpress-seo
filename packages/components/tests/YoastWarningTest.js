import React from "react";
import renderer from "react-test-renderer";
import Warning from "../src/Warning";
import { ThemeProvider } from "styled-components";

test( "YoastWarning matches the snapshot", () => {
	/* eslint-disable react/jsx-no-target-blank */
	const component = renderer.create(
		<Warning
			message={ [
				"This is a warning message",
				<a key={ 1 } href="https://yoast.com" target="_blank">Go to Yoast.com</a>,
				<p key={ 2 }>React will render this</p>,
			] }
		/>
	);
	/* eslint-enable react/jsx-no-target-blank */

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "YoastWarning does not render without a message", () => {
	const component = renderer.create(
		<Warning />
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
	expect( tree ).toBeNull();
} );

test( "YoastWarning with a right to left language matches the snapshot", () => {
	/* eslint-disable react/jsx-no-target-blank */
	const component = renderer.create(
		<ThemeProvider theme={ { isRtl: true } }>
			<Warning
				message={ [
					"This is a warning message",
					<a key={ 1 } href="https://yoast.com" target="_blank">Go to Yoast.com</a>,
					<p key={ 2 }>React will render this</p>,
				] }
			/>
		</ThemeProvider>
	);
	/* eslint-enable react/jsx-no-target-blank */

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
