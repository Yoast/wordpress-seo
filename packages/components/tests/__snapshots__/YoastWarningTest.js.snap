// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YoastWarning does not render without a message 1`] = `null`;

exports[`YoastWarning matches the snapshot 1`] = `
.c1 {
  width: 16px;
  height: 16px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 16px;
  background: #fff3cd;
  color: #674e00;
}

.c2 {
  margin-top: 2px;
}

.c3 {
  margin: 0 0 0 8px;
}

<div
  className="c0"
>
  <svg
    aria-hidden={true}
    className="c1 yoast-svg-icon yoast-svg-icon-exclamation-triangle c2"
    fill="currentColor"
    focusable="false"
    role="img"
    size="16px"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1024 1375v-190q0-14-9.5-23.5T992 1152H800q-13 0-22.5 9.5T768 1185v190q0 14 9.5 23.5t22.5 9.5h192q13 0 22.5-9.5t9.5-23.5zm-2-374l18-459q0-12-10-19-13-11-24-11H786q-11 0-24 11-10 7-10 21l17 457q0 10 10 16.5t24 6.5h185q14 0 23.5-6.5t10.5-16.5zm-14-934l768 1408q35 63-2 126-17 29-46.5 46t-63.5 17H128q-34 0-63.5-17T18 1601q-37-63-2-126L784 67q17-31 47-49t65-18 65 18 47 49z"
    />
  </svg>
  <div
    className="c3"
  >
    This is a warning message
    <a
      href="https://yoast.com"
      target="_blank"
    >
      Go to Yoast.com
    </a>
    <p>
      React will render this
    </p>
  </div>
</div>
`;

exports[`YoastWarning with a right to left language matches the snapshot 1`] = `
.c1 {
  width: 16px;
  height: 16px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 16px;
  background: #fff3cd;
  color: #674e00;
}

.c2 {
  margin-top: 2px;
}

.c3 {
  margin: 0 8px 0 0;
}

<div
  className="c0"
>
  <svg
    aria-hidden={true}
    className="c1 yoast-svg-icon yoast-svg-icon-exclamation-triangle c2"
    fill="currentColor"
    focusable="false"
    role="img"
    size="16px"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1024 1375v-190q0-14-9.5-23.5T992 1152H800q-13 0-22.5 9.5T768 1185v190q0 14 9.5 23.5t22.5 9.5h192q13 0 22.5-9.5t9.5-23.5zm-2-374l18-459q0-12-10-19-13-11-24-11H786q-11 0-24 11-10 7-10 21l17 457q0 10 10 16.5t24 6.5h185q14 0 23.5-6.5t10.5-16.5zm-14-934l768 1408q35 63-2 126-17 29-46.5 46t-63.5 17H128q-34 0-63.5-17T18 1601q-37-63-2-126L784 67q17-31 47-49t65-18 65 18 47 49z"
    />
  </svg>
  <div
    className="c3"
  >
    This is a warning message
    <a
      href="https://yoast.com"
      target="_blank"
    >
      Go to Yoast.com
    </a>
    <p>
      React will render this
    </p>
  </div>
</div>
`;
